tests/test_trainer.py

```diff
@@ -62,7 +62,7 @@ def test_trainer():
 
     results = initialize_trainer(CONFIG)
     print(results)
-    assert results[0]["test_acc"] > 0.6
+    assert results[0]["test_acc"] > 0.3
```

Comment:
Was this giving a value much below 0.6 for you? It's giving 0.6992337703704834 deteministically when I run it on my mac, although it isn't really trained on much.

Reply:
Weird. Yeah it was giving me like 37%. See screen shot. There was some additional output as well (pasted below). 

<img width="977" alt="Screen Shot 2021-07-03 at 8 45 36 PM" src="https://user-images.githubusercontent.com/1328807/124369959-c587e500-dc3f-11eb-8f63-9e6511228289.png">



```
tests/test_trainer.py:65: AssertionError
----------------------------------- Captured stdout call -----------------------------------
Epoch 0: 100%|██████████| 4/4 [00:34<00:00,  8.70s/it, loss=0.719, v_num=3]
Testing: 100%|██████████| 26/26 [00:25<00:00,  1.03it/s] 
--------------------------------------------------------------------------------
DATALOADER:0 TEST RESULTS
{'test_acc': 0.3728836178779602, 'test_loss': 0.5584720969200134}
--------------------------------------------------------------------------------
[{'test_loss': 0.5584720969200134, 'test_acc': 0.3728836178779602}]
----------------------------------- Captured stderr call -----------------------------------
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForSequenceClassification: ['cls.predictions.transform.dense.weight', 'cls.predictions.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.decoder.weight', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.bias', 'cls.seq_relationship.weight', 'cls.predictions.transform.LayerNorm.bias']
- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForSequenceClassification were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
------------------------------------ Captured log call -------------------------------------
INFO     pytorch_lightning.utilities.seed:seed.py:65 Global seed set to 1234
INFO     pytorch_lightning.utilities.distributed:accelerator_connector.py:591 GPU available: False, used: False
INFO     pytorch_lightning.utilities.distributed:accelerator_connector.py:593 TPU available: False, using: 0 TPU cores
INFO     pytorch_lightning.core.lightning:lightning.py:1496 
  | Name  | Type                          | Params
--------------------------------------------------------
0 | model | BertForSequenceClassification | 109 M 
--------------------------------------------------------
109 M     Trainable params
0         Non-trainable params
109 M     Total params
437.947   Total estimated model params size (MB)
INFO     pytorch_lightning.utilities.seed:seed.py:65 Global seed set to 1234
```

Reply:
Strange... @laurahanu is a high value necessarily expected here?

Reply:
FYI - I temporarily changed back to 0.6 to show the failing test on CI. You can see it here:

https://github.com/unitaryai/detoxify/pull/20/checks?check_run_id=2984564970#step:8:278

Now I'm changing it to 0.3, so tests will pass.

Reply:
Interesting, checked that it is also [failing with CI on master](https://github.com/unitaryai/detoxify/pull/23). I tried upgrading to `torch==1.9.0` `pytorch-lightning==1.3.8` on my laptop and it was still passing (0.699992835521698) but then also upgrading `transformers` (from 3.4.0 to 4.8.2) it fails (0.3728836178779602). Hopefully just random seed variability, but we should check the models are still working as expected with the newer versions of `transformers`.

Reply:
The 0.6 was chosen somewhat arbitrarily so it shouldn't matter that much if we change it to a lower value. I've checked and we still get the same values with the same input with the newest `torch==1.9.0` and `transformers==4.8.2` so the models should be working as expected.

![image](https://user-images.githubusercontent.com/32672979/124498208-e07b6600-ddb3-11eb-895e-1cbae4013c94.png)
![image](https://user-images.githubusercontent.com/32672979/124498255-f12bdc00-ddb3-11eb-9e39-89a2e1c54259.png)

Seems like the culprit here is [`transformers==4.6.0`](https://github.com/huggingface/transformers/releases/tag/v4.6.0) with this [specific PR](https://github.com/huggingface/transformers/pull/11471) where they introduce a fast initialisation of BERT. To get the same results as before we can just set `_fast_init=False` in:

```python
model = BertForSequenceClassification.from_pretrained("bert-base-cased", _fast_init=False)
```